generator/template.go

```diff
@@ -229,20 +353,39 @@ func fieldName(r *registry.Registry) func(name string) string {
 func renderURL(r *registry.Registry) func(method data.Method) string {
 	fieldNameFn := fieldName(r)
 	return func(method data.Method) string {
-		url := method.URL
+		methodURL := method.URL
 		reg := regexp.MustCompile("{([^}]+)}")
-		matches := reg.FindAllStringSubmatch(url, -1)
+		matches := reg.FindAllStringSubmatch(methodURL, -1)
+		fieldsInPath := "["
 		if len(matches) > 0 {
 			log.Debugf("url matches %v", matches)
 			for _, m := range matches {
 				expToReplace := m[0]
 				fieldName := fieldNameFn(m[1])
 				part := fmt.Sprintf(`${req["%s"]}`, fieldName)
-				url = strings.ReplaceAll(url, expToReplace, part)
+				methodURL = strings.ReplaceAll(methodURL, expToReplace, part)
+				fieldsInPath += fmt.Sprintf(`"%s", `, fieldName)
+			}
+		}
+		fieldsInPath = strings.TrimRight(fieldsInPath, ", ") + "]"
+
+		if !method.ClientStreaming && method.HTTPMethod == "GET" {
+			// parse the url to check for query string
+			parsedURL, err := url.Parse(methodURL)
+			if err != nil {
+				return methodURL
+			}
+			renderURLSearchParamsFn := fmt.Sprintf("${fm.renderURLSearchParams(req, %s)}", fieldsInPath)
+			// prepend "&" if query string is present otherwise prepend "?"
+			// trim leading "&" if present before prepending it
+			if parsedURL.RawQuery != "" {
+				methodURL += "&" + strings.TrimRight(renderURLSearchParamsFn, "&")
```

Comment:
Removing leading `&` if present in URL path here.

Reply:
shouldn't this be Trimleft?

Reply:
ah I am sorry since you mentioned that URL may contain "&" I think it should be 
`methodURL = strings.TrimRight(methodURL, "&") + strings.TrimLeft(renderURLSearchParamsFn, "&")`

Reply:
it's for the case where methodURL ends with `&`, 

so it should be `strings.TrimRight(methodURL, "&")  + "&" + renderURLSearchParamsFn.

you are missing the `&` in the middle.

Reply:
Sorry about the many mistakes and typos. Will fix this 🙇🏼